### ISSUE TITLE
Add additional password policy notifcation 

### ISSUE THREAD

**Issue by mmzeeman** [zotonic/zotonic#1732]
In order to implement additional password policy rules it would be handy to add a new notification. This notification can be used to add site specific password checks like the inclusion of special characters, one or more digits and such.

The record can be made like this.

```erlang
-record(identity_password_policy, {rsc_id, new_password}).
```

When a password is changed, a notification is tried. Custom code can check additional password policy rules. When `undefined` or `ok` is returned nothing happens. When an error occurs it can return the an error reason which can be picked up in the logon template which can provide a custom error message.

**Comment by mworrell** [comment]
As you also want to store the last *N* passwords, do we need any additional notifications for identity changes?   (Preferably in the database transaction modifying the identity)

**Comment by mmzeeman** [comment]
Yes, I also need a notification in that location to audit password resets and changes.

**Comment by mworrell** [comment]
We have now extra password checks.
